### PR TITLE
feat: redefinition purse

### PR DESCRIPTION
**Pull request by nowooj**
- Purse divided into reward and commission
- Add to community purse.

P.S.
--
Only redefine purse. Next pr I'll add to this amount in distribution(economy)

### CODE REVIEW

File: execution-engine/engine-tests/src/test/hdac_system_contracts/pop_install.rs [hunk 1]
```diff
@@ -17,11 +17,13 @@ const DEPLOY_HASH_2: [u8; 32] = [2u8; 32];
 const N_VALIDATORS: u8 = 5;
 
 // one named_key for each validator and four for the purses and the total supply amount.
-const EXPECTED_KNOWN_KEYS_LEN: usize = ((N_VALIDATORS * 2) as usize) + 4;
+const EXPECTED_KNOWN_KEYS_LEN: usize = ((N_VALIDATORS * 2) as usize) + 5 + 1;
```
Comment (jhyeom26) [inline execution-engine/engine-tests/src/test/hdac_system_contracts/pop_install.rs]: Seems to need to correct the comment accodingly

File: execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs [hunk 2]
```diff
@@ -274,8 +274,29 @@ impl ProofOfProfessionContract {
...
+            / U512::from(100);
+
+        if total != (reward_amount + commission_amount) {
+            let remain_amount = total - reward_amount - commission_amount;
+
+            let communtiy_purse =
+                get_purse(uref_names::POS_COMMUNITY_PURSE).map_err(PurseLookupError::communtiy)?;
+
+            system::transfer_from_purse_to_purse(payment_purse, communtiy_purse, remain_amount)
+                .map_err(|_| Error::FailedTransferToRewardsPurse)?;
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs]: It seems like a wrong error, `Error::FailedTransferToRewardsPurse` -> `Error::FailedTransferToCommunityPurse`

File: execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs [hunk 3]
```diff
@@ -274,8 +274,29 @@ impl ProofOfProfessionContract {
...
+        }
+
+        system::transfer_from_purse_to_purse(payment_purse, reward_purse, reward_amount)
+            .map_err(|_| Error::FailedTransferToRewardsPurse)?;
+
+        system::transfer_from_purse_to_purse(payment_purse, commission_purse, commission_amount)
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs:299]: It seems like a wrong error, `Error::FailedTransferToRewardsPurse` -> `Error::FailedTransferToCommissionPurse`

Comment (nowooj) [reply execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs:299]: https://github.com/hdac-io/CasperLabs/pull/30/commits/653d226b40ca1529d6d067b382f03e42199c967d

File: execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs (hunk 2 above)
Comment (nowooj) [reply execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs]: https://github.com/hdac-io/CasperLabs/pull/30/commits/653d226b40ca1529d6d067b382f03e42199c967d

File: execution-engine/engine-tests/src/test/hdac_system_contracts/pop_install.rs (hunk 1 above)
Comment (nowooj) [reply execution-engine/engine-tests/src/test/hdac_system_contracts/pop_install.rs]: https://github.com/hdac-io/CasperLabs/pull/30/commits/653d226b40ca1529d6d067b382f03e42199c967d

**Review by psy2848048 — APPROVED** [review]

**Review by jhyeom26 — APPROVED** [review]

**Pull request merged**